### PR TITLE
CSHARP-4140: Only use $getField when absolutely necessary.

### PR DESCRIPTION
**Pull request by rstam**

### CODE REVIEW

File: src/MongoDB.Driver/Linq/Linq3Implementation/Ast/Expressions/AstGetFieldExpression.cs [hunk 1]
```diff
@@ -67,17 +67,14 @@ public bool HasSafeFieldName(out string fieldName)
...
                 {
                     return true;
                 }
             }
 
             fieldName = null;
             return false;
-
-            static bool IsSafeFirstChar(char c) => c == '_' || (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z');
-            static bool IsSafeSubsequentChar(char c) => IsSafeFirstChar(c) || (c >= '0' && c <= '9');
```
Comment (rstam) [inline src/MongoDB.Driver/Linq/Linq3Implementation/Ast/Expressions/AstGetFieldExpression.cs:80]: These look like the rules for server variable names, not fields!

File: src/MongoDB.Driver/Linq/Linq3Implementation/Ast/Expressions/AstGetFieldExpression.cs [hunk 2]
```diff
@@ -67,17 +67,14 @@ public bool HasSafeFieldName(out string fieldName)
                 constantFieldName.Value is BsonString stringfieldName)
             {
                 fieldName = stringfieldName.Value;
-                if (fieldName.Length > 0 && IsSafeFirstChar(fieldName[0]) && fieldName.Skip(1).All(c => IsSafeSubsequentChar(c)))
+                if (fieldName.Length > 0 && fieldName[0] != '$' && !fieldName.Contains('.'))
```
Comment (rstam) [inline src/MongoDB.Driver/Linq/Linq3Implementation/Ast/Expressions/AstGetFieldExpression.cs:70]: As far as I can tell $getField is only required when the field name starts with `$` or contains `.`.

File: tests/MongoDB.Driver.Tests/Linq/Linq3ImplementationTests/Jira/CSharp4140Tests.cs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+using Xunit;
+
+namespace MongoDB.Driver.Tests.Linq.Linq3ImplementationTests.Jira
+{
+    public class CSharp4140Tests : Linq3IntegrationTest
+    {
+        [Theory]
+        [InlineData("abc")]
+        [InlineData("123")]
+        [InlineData("12$")]
```
Comment (rstam) [inline tests/MongoDB.Driver.Tests/Linq/Linq3ImplementationTests/Jira/CSharp4140Tests.cs:28]: Non-leading `$` do *not* require using $getField

File: tests/MongoDB.Driver.Tests/Linq/Linq3ImplementationTests/Jira/CSharp4140Tests.cs [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        [InlineData("!@#")]
```
Comment (rstam) [inline tests/MongoDB.Driver.Tests/Linq/Linq3ImplementationTests/Jira/CSharp4140Tests.cs:29]: Surprisingly this is a valid field name.

File: tests/MongoDB.Driver.Tests/Linq/Linq3ImplementationTests/Jira/CSharp4140Tests.cs [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+            var stages = Translate(collection, queryable);
+            AssertStages(stages, $"{{ $project : {{ _v : '${fieldName}', _id : 0 }} }}");
+
+            var results = queryable.ToList();
+            results.Should().Equal(123);
+        }
+
+        [Theory]
+        [InlineData("")]
```
Comment (rstam) [inline tests/MongoDB.Driver.Tests/Linq/Linq3ImplementationTests/Jira/CSharp4140Tests.cs:45]: Also surprisingly a valid field name, though it does require using $getField.

**Review by JamesKovacs — APPROVED** [review]
LGTM

**Pull request merged**